pumps/graph_mongo.go

```diff
@@ -0,0 +1,145 @@
+package pumps
+
+import (
+	"context"
+	"strings"
+
+	"github.com/TykTechnologies/tyk-pump/analytics"
+	"github.com/mitchellh/mapstructure"
+	"github.com/sirupsen/logrus"
+)
+
+const mongoGraphPrefix = "mongo-graph-pump"
+
+type GraphMongoPump struct {
+	CommonPumpConfig
+	MongoPump
+}
+
+func (g *GraphMongoPump) New() Pump {
+	return &GraphMongoPump{}
+}
+
+func (g *GraphMongoPump) GetEnvPrefix() string {
+	return g.dbConf.EnvPrefix
+}
+
+func (g *GraphMongoPump) GetName() string {
+	return "MongoDB Graph Pump"
+}
+
+func (g *GraphMongoPump) Init(config interface{}) error {
+	g.dbConf = &MongoConf{}
+	g.log = log.WithField("prefix", mongoGraphPrefix)
+	g.MongoPump.CommonPumpConfig = g.CommonPumpConfig
+
+	err := mapstructure.Decode(config, &g.dbConf)
+	if err == nil {
+		err = mapstructure.Decode(config, &g.dbConf.BaseMongoConf)
+		g.log.WithFields(logrus.Fields{
+			"url":             g.dbConf.GetBlurredURL(),
+			"collection_name": g.dbConf.CollectionName,
+		}).Info("Init")
+		if err != nil {
+			panic(g.dbConf.BaseMongoConf)
+		}
+	}
+	if err != nil {
```

Comment:
Checking an error and breaking the flow if it is a non-nil look more idiomatic. 

```go
if err != nil {
   return fmt.Errorf("error while decode configuration: %w", err)
}

err = mapstructure.Decode(config, &g.dbConf.BaseMongoConf)
g.log.WithFields(logrus.Fields{
	"url":             g.dbConf.GetBlurredURL(),
	"collection_name": g.dbConf.CollectionName,
}).Info("Init")
if err != nil {
	panic(g.dbConf.BaseMongoConf)
}
```

Caller of `Init` should log this error, I think.

Reply:
Yeah i initially copied the current mongo implementation but changed the structure and logic to return an error rather than panic, the caller logs this error and skips the pump initialization